src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4613,7 +4613,11 @@ private static bool IsTargetTypedExpression(BoundExpression node)
             {
                 var (returnExpr, resultType, _) = returns[i];
                 resultTypes.Add(resultType);
-                placeholdersBuilder.Add(CreatePlaceholderIfNecessary(returnExpr, resultType));
+
+                if (!IsTargetTypedExpression(returnExpr))
```

Comment:
The tests seem to focus on collection expressions as the "target-typed expression". Consider adding tests for the other kinds of target-typed expressions (null, default, switches, conditionals, lambdas, tuples, ...).

Reply:
the scenario is about when:
1. a target-typed expression is subject to best-type analysis (e.g. `b ? ["1"] : new[] { "2" }`),
2. *and*, there is a nested nullability reinference occurring in the other best-type candidate expressions, which we want to "win" over any initial-bound type which the target-typed expression may have.

So, there might be an interesting scenario to test like in `b ? ("1", "2") : GenericMethod("1")`, or something like that.

Reply:
Have added a test based on this suggestion.

Reply:
I didn't follow. The test you added doesn't have lambdas.

Reply:
There are three places where an `IsTargetTypedExpression` check was added: conditionals, lambdas and switches. The added test seems to cover conditionals. 
Consider adding a similar test showing the effects of the fix on other target-typed expressions besides collection expressions in lambdas and switches.

Reply:
Switches are covered in `TargetTypedExpressions_NestedNullability`.
Lambdas are covered in `CollectionExpression_NestedNullability_06`. Note that one of the cases involves collection-exprs, and the other case doesn't involve collection-exprs, but still shows an incorrect behavior, tracked in #74609